syntax = "proto3";

package routing;

service RoutingService {
  rpc GetRoute (RouteRequest) returns (RouteResponse) {}
}

message RouteRequest {
  Point origin = 1;
  Point destination = 2;
  string mode = 3;
}

message RouteResponse {
  repeated RouteStep steps = 1;
  double total_distance_meters = 2;
  double total_duration_seconds = 3;
}

message Point {
  double latitude = 1;
  double longitude = 2;
}

message RouteStep {
  string instruction = 1;
  double distance_meters = 2;
  double duration_seconds = 3;
  Point start_location = 4;
  Point end_location = 5;
  string type = 6;
  string line_name = 7;
}
